src/dirs.proj

```diff
@@ -15,6 +15,10 @@
     <Project Include="shims\ApiCompat.proj" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(DirectoryToBuild)' != ''">
+    <Project Include="$(DirectoryToBuild)/**/*.csproj" />
```

Comment:
These projects are serialized (i.e. SerializeProjects=true) so there shouldn't be any races.